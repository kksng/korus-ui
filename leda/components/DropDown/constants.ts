export enum InteractionModes {
  Click = 'click'
}
