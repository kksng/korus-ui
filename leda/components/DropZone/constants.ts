export const EMPTY_DROP_ZONE_FILES = {
  acceptedFiles: [],
  rejectedFiles: [],
};
