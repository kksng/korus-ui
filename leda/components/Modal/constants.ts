export const WINDOW_SIZES = {
  sm: 480,
  md: 600,
  lg: 800,
} as const;
