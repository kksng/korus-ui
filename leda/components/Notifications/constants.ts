export const DEFAULT_NOTIFICATION_LIFETIME = 5000; // milliseconds

export enum ChangeMethods {
  CloseIconClick = 'close-icon-click',
  Delay = 'delay'
}
