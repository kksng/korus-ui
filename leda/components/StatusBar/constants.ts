export const STEP_POSITION = {
  CURRENT: 'current',
  NEXT: 'next',
  PREV: 'prev',
} as const;

export const STEP_TYPES = {
  DANGER: 'danger',
  PROGRESS: 'progress',
  SUCCESS: 'success',
} as const;

export const DATA_TYPES = {
  OBJECT: 'object',
  STRING: 'string',
} as const;
