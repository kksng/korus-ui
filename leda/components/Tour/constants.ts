export const Position = {
  Bottom: 'bottom-center',
  BottomLeft: 'bottom-left',
  BottomRight: 'bottom',
  Left: 'left',
  Right: 'right',
  Top: 'top-center',
  TopLeft: 'top-left',
  TopRight: 'top',
} as const;
