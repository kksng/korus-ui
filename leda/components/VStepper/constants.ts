export enum Display {
  Block = 'block',
  None = 'none'
}
