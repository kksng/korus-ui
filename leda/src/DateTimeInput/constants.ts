export const COMPONENT_TYPES = {
  DATE_ONLY: 'date-only',
  TIME_ONLY: 'time-only',
  DATE_TIME: 'date-time',
} as const;

export const HOURS_LIMITS = [0, 23];

export const MINUTES_LIMITS = [0, 59];

export const MONTHS_IN_ROW = 4;

export const YEARS_IN_ROW = 4;

export const DAYS_IN_WEEK = 7;

export const KEYS = {
  LEFT: 'ArrowLeft',
  LEFT_IE: 'Left',
  RIGHT: 'ArrowRight',
  RIGHT_IE: 'Right',
  UP: 'ArrowUp',
  UP_IE: 'Up',
  DOWN: 'ArrowDown',
  DOWN_IE: 'Down',
  ENTER: 'Enter',
  ESC: 'Escape',
  ESC_IE: 'Esc',
  TAB: 'Tab',
} as const;
