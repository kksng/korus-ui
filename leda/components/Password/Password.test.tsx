import React from 'react';
import {
  render,
  fireEvent,
} from '@testing-library/react';

import userEvent from '@testing-library/user-event';
import { Password } from './index';
import { PasswordStrength } from './constants';
import { validatePassword } from '../../validators/password';

const validName = 'test';
const validValue = '123';
const on = true;
const off = false;

describe('Password snapshots collection', () => {
  test('is Password component render right?', () => {
    const { container } = render(<Password name={validName} />);

    expect(container.firstChild)
      .toMatchSnapshot();
  });
  test('is Password render right with defaultValue?', () => {
    const { container } = render(<Password defaultValue={validValue} name={validName} />);
    const input = container.querySelectorAll('input.password-input')[0];

    expect(input)
      .toHaveAttribute('value', validValue);

    expect(container.firstChild)
      .toMatchSnapshot();
  });
  test('is Password render right with hasClearButton?', () => {
    const { container, rerender } = render(<Password defaultValue={validValue} hasClearButton={on} name={validName} />);
    const icon = container.querySelectorAll('.password-clear-icon')[0];
    const input = container.querySelectorAll('input.password-input')[0];

    expect(icon)
      .toBeInTheDocument();

    expect(input)
      .toBeInTheDocument();

    expect(container.firstChild)
      .toMatchSnapshot();

    rerender(<Password defaultValue={validValue} hasClearButton={off} name={validName} />);

    expect(icon)
      .not
      .toBeInTheDocument();
  });
});
describe('Password attributes test collection', () => {
  test('is Password work right with isDisabled attributes?', () => {
    const onFocus = jest.fn();
    const { container } = render(<Password isDisabled onFocus={onFocus} />);
    const input = container.querySelectorAll('input.password-input')[0] as HTMLInputElement | undefined;

    expect(container.querySelectorAll('.password-element-wrapper.disabled'))
      .toHaveLength(1);

    expect(input)
      .toHaveAttribute('disabled');

    input?.focus();

    expect(onFocus)
      .toHaveBeenCalledTimes(0);
  });
  test('is Password work right with letterCase attributest?', () => {
    const onChange = jest.fn();
    const inputUpper = 'upper';
    const inputLower = 'lower';
    const { container, rerender } = render(<Password letterCase={inputUpper} onChange={onChange} />);
    const input = container.querySelectorAll('input.password-input')[0];

    userEvent.type(input, 'abc');

    expect(onChange)
      .lastCalledWith(expect.objectContaining({
        component: expect.objectContaining({
          value: 'ABC',
        }),
      }));

    rerender(<Password letterCase={inputLower} onChange={onChange} />);

    userEvent.type(input, 'ABC');

    expect(onChange)
      .lastCalledWith(expect.objectContaining({
        component: expect.objectContaining({
          value: 'abc',
        }),
      }));
  });
  test('is Password work right with maxLength attributes?', () => {
    const max = 2;
    const onChange = jest.fn();
    const { container } = render(<Password maxLength={max} onChange={onChange} />);
    const input = container.querySelectorAll('input.password-input')[0];

    userEvent.type(input, 'abcd');

    expect(onChange)
      .lastCalledWith(expect.objectContaining({
        component: expect.objectContaining({
          value: 'ab',
        }),
      }));
  });
  test('is Password work right with allowedSymbols attributes?', () => {
    const onlyNumber = new RegExp(/^\d+$/);
    const onChange = jest.fn();
    const { container } = render(<Password allowedSymbols={onlyNumber} onChange={onChange} />);
    const input = container.querySelectorAll('input.password-input')[0];

    userEvent.type(input, '0abcd');

    expect(onChange)
      .lastCalledWith(expect.objectContaining({
        component: expect.objectContaining({
          value: '0',
        }),
      }));
  });
  test('is Password work right with forbiddenSymbols attributes?', () => {
    const onlyNumber = new RegExp(/^\d+$/);
    const onChange = jest.fn();
    const { container } = render(<Password forbiddenSymbols={onlyNumber} onChange={onChange} />);
    const input = container.querySelectorAll('input.password-input')[0];

    userEvent.type(input, 'a0123');

    expect(onChange)
      .lastCalledWith(expect.objectContaining({
        component: expect.objectContaining({
          value: 'a',
        }),
      }));
  });
  test('is Password work right with minPasswordEvaluationLength attributes?', () => {
    const emptyPasswordMessage = 'Используйте строчные и прописные латинские буквы и цифры, не менее 8 символов';
    const notEmptyPasswordMessage = 'Слабый пароль';
    const minEvalLength = 4;
    const {
      container,
      getByText,
      rerender,
    } = render(<Password minPasswordEvaluationLength={minEvalLength} />);
    const input = container.querySelectorAll('input.password-input')[0];

    userEvent.type(input, '012');

    expect(getByText(emptyPasswordMessage))
      .toBeInTheDocument();

    rerender(<Password minPasswordEvaluationLength={minEvalLength} />);

    userEvent.type(input, '0123');

    expect(getByText(notEmptyPasswordMessage))
      .toBeInTheDocument();
  });
  test('is Password work right with passwordEvalutors attributes work right?', () => {
    const passworLowTxt = 'Пароль слабоват';
    const passwordMediumTxt = 'Надёжный пароль';
    const passwordStrongTxt = 'Отличный пароль';
    const passwordEvaluators = [
      {
        strengthLevel: PasswordStrength.Low,
        evaluator: () => true,
        evaluationMessage: passworLowTxt,
      },
      {
        strengthLevel: PasswordStrength.Medium,
        evaluator: (password: string) => {
          if (validatePassword(password)) return true;
          return false;
        },
        evaluationMessage: passwordMediumTxt,
      },
      {
        strengthLevel: PasswordStrength.Strong,
        evaluator: (password: string) => {
          if (
            validatePassword(password)
            && password.length > 12
            && /[!@#$%*]/.test(password)
          ) {
            return true;
          }
          return false;
        },
        evaluationMessage: passwordStrongTxt,
      },
    ];
    const {
      container,
      getByText,
    } = render(<Password passwordEvaluators={passwordEvaluators} />);

    const input = container.querySelectorAll('input.password-input')[0];

    userEvent.type(input, '3233');

    expect(getByText(passworLowTxt))
      .toBeInTheDocument();

    userEvent.type(input, '3233@#Dq');

    expect(getByText(passwordMediumTxt))
      .toBeInTheDocument();

    userEvent.type(input, '3233@#DqEErrq');

    expect(getByText(passwordStrongTxt))
      .toBeInTheDocument();
  });
});
describe('Password events test collection', () => {
  test('is Password work right with onBlur event listner?', () => {
    const onBlur = jest.fn();
    const { container } = render(<Password defaultValue={validValue} name={validName} onBlur={onBlur} />);
    const input = container.querySelectorAll('input.password-input')[0];

    fireEvent.blur(input);

    expect(input)
      .not
      .toHaveFocus();

    expect(onBlur)
      .toBeCalledTimes(1);

    expect(onBlur)
      .lastCalledWith(expect.objectContaining({
        component: expect.objectContaining({
          name: validName,
        }),
      }));
  });
  test('is Password work right with onFocus event listener?', () => {
    const onFocus = jest.fn();
    const { container } = render(<Password defaultValue={validValue} name={validName} onFocus={onFocus} />);
    const input = container.querySelectorAll('input.password-input')[0];

    fireEvent.focus(input);

    expect(onFocus)
      .toBeCalledTimes(1);

    expect(onFocus)
      .lastCalledWith(expect.objectContaining({
        component: expect.objectContaining({
          name: validName,
        }),
      }));
  });
  test('is Password work right with onChange event listener?', () => {
    const onChange = jest.fn();
    const { container } = render(<Password defaultValue={validValue} name={validName} onChange={onChange} />);
    const input = container.querySelectorAll('input.password-input')[0];

    userEvent.type(input, validValue);

    expect(onChange)
      .toBeCalledTimes(validValue.length);

    expect(onChange)
      .lastCalledWith(expect.objectContaining({
        component: expect.objectContaining({
          name: validName,
          value: validValue,
        }),
      }));
  });
});
