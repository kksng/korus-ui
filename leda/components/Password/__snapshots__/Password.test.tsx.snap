// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password snapshots collection is Password component render right? 1`] = `
<div
  class="password-wrapper"
>
  <div
    class="password-element-wrapper"
  >
    <input
      aria-invalid="false"
      class="password-input"
      name="test"
      type="password"
      value=""
    />
    <i
      class="password-is-hidden"
    />
  </div>
  <div
    class="password-message-default"
  >
    Используйте строчные и прописные латинские буквы и цифры, не менее 8 символов
  </div>
</div>
`;

exports[`Password snapshots collection is Password render right with defaultValue? 1`] = `
<div
  class="password-wrapper"
>
  <div
    class="password-element-wrapper"
  >
    <input
      aria-invalid="false"
      class="password-input"
      name="test"
      type="password"
      value="123"
    />
    <i
      class="password-is-hidden"
    />
  </div>
  <div
    class="password-message-default"
  >
    Используйте строчные и прописные латинские буквы и цифры, не менее 8 символов
  </div>
</div>
`;

exports[`Password snapshots collection is Password render right with hasClearButton? 1`] = `
<div
  class="password-wrapper"
>
  <div
    class="password-element-wrapper"
  >
    <input
      aria-invalid="false"
      class="password-input"
      name="test"
      type="password"
      value="123"
    />
    <i
      class="password-clear-icon"
    />
    <i
      class="password-is-hidden"
    />
  </div>
  <div
    class="password-message-default"
  >
    Используйте строчные и прописные латинские буквы и цифры, не менее 8 символов
  </div>
</div>
`;
